### ISSUE TITLE
Out of memory exception when loading large repo

### ISSUE THREAD

**Issue by athiruban** [jughyd/GitFx#14]
When I open JGit repo in GitFx, it is failing with out of memory exception.

Even if we select only first 100 revisions it is still exhibiting the same error,
GitRepoMetaData > getShortMessage() -> 
while ((i < 100) && treeWalk.next()) {
commitFiles.add(treeWalk.getPathString()); i++ }

As Rohit mentioned this error comes from JGit internals.

**Comment by rajmahendra** [comment]
Is this issue still occurring?

**Comment by athiruban** [comment]
I just tried adding jgit repo. It clocked for a while and gitfx app is not showing the repo as added to it.

Is it working fine in your machine?

**Comment by rohitvvv** [comment]
Hey @athiruban thanks for confirming. These issues become important as we try to open bigger repositories and hit close to beta release. Currently we are following a very crude strategy to hold everything in memory and that's the reason you may be seeing these issues. I currently am not having any issues with opening GitFx repo. However I am sure as I open bigger ones, we will run short of heap. 
The solution is to read chunks of data as required and not load everything in memory. My focus is currently to have one full iteration of GitFx working. Will attend to this one latter. Feel free to comment with your suggestions on fixing this issue.

**Comment by athiruban** [comment]
You are correct @rohitvvv instead of loading everything in memory we need to read chunks of data. We need a queuing facility to git rid of this issue. I will research more on this and share you the details.
